test/sqllogictest/explain/decorrelated_plan_as_text.slt

```diff
@@ -26,131 +29,1035 @@ SELECT * FROM t WHERE a IS NOT NULL
 
 mode cockroach
 
+# Test constant error.
 query T multiline
 EXPLAIN DECORRELATED PLAN AS TEXT FOR
-SELECT
-  *
-FROM
-  T as X
-WHERE
-  NOT EXISTS (SELECT * FROM T as Y WHERE X.a = Y.b)
-LIMIT 10
+SELECT 1 / 0
+----
+Let
+  Project #0
+    Map (1 / 0)
+      Let
+        Get l1
+        Where
+          l1 =
+            CrossJoin
+              Get l0
+              Constant
+                - ()
+  Where
+    l0 =
+      Constant
+        - ()
+
+EOF
+
+# Test constant with two elements.
+query T multiline
+EXPLAIN DECORRELATED PLAN AS TEXT FOR
+(SELECT 1, 2) UNION ALL (SELECT 1, 2) UNION ALL (SELECT 3, 4)
+----
+Let
+  Union
+    Project #2, #3
+      Project #0..=#3
+        Map #1
+          Map #0
+            Let
+              Get l5
+              Where
+                l5 =
+                  Union
+                    Project #2, #3
+                      Project #0..=#3
+                        Map #1
+                          Map #0
+                            Let
+                              Get l2
+                              Where
+                                l2 =
+                                  Project #0, #1
+                                    Map 2
+                                      Map 1
+                                        Let
+                                          Get l1
+                                          Where
+                                            l1 =
+                                              CrossJoin
+                                                Get l0
+                                                Constant
+                                                  - ()
+                    Project #2, #3
+                      Project #0..=#3
```

Comment:
I'm not particularly fond of doing that at the moment for several reasons:

- It makes rendering context sensitive (need to output a `- ` prefix if the current node is a child of a `Union` or `Join` and an empty prefix otherwise).
- It means that we are not consistent in the rendering of our inputs.
- It won't help a lot for plans that have unions of joins or joins of unions.

However, I agree in principle that the output of decorrelated plans can be improved. I've tried to address the issue by adding a transform that normalizes the `Let` nodes in a given tree, pulling all let bindings to the top of the tree. This leads to "flatter" decorrelated plans because we render a chain of nested let bindings as a single let note with multiple bindings.

I'm happy to revisit this decision after we have all outputs covered and have used the format for a while.